### PR TITLE
allow generic string as metric name

### PR DESCRIPTION
**Pull request by noamokman**
- allow generic string as metric name
- .

### CODE REVIEW

File: src/monitor.ts [hunk 1]
```diff
@@ -115,11 +115,15 @@ const innerMonitor = <Callable, Scope extends string, Method extends string>({
 
 export const createMonitor =
   <Scope extends string>({ scope, ...initOptions }: InitOptions<Scope>) =>
-  <Callable, Method extends string>(method: CamelCase<Method>, callable: () => Callable, options?: MonitorOptions<Callable>) =>
+  <Callable, Method extends string>(
+    method: CamelCase<Method> extends never ? string : CamelCase<Method>,
```
Comment (SharonGrossman) [inline src/monitor.ts:119]: extends never ? : string
mind explaning?
CamelCase<Method> didn't let you put a variable as a scope / method name?

Comment (noamokman) [reply src/monitor.ts:119]: Yeah, it resolves to never, so this is the workaround

Comment (noamokman) [reply src/monitor.ts:119]: Also, look at the test I added, it failed before

**Review by SharonGrossman — APPROVED** [review]

**Comment by SharonGrossman** [comment]
package.json version bump

**Comment by noamokman** [comment]
@SharonGrossman thanks!

**Pull request merged**